framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -145,12 +147,19 @@ public ValueAnnotatedTypeFactory(BaseTypeChecker checker) {
         // this line just registers the alias. The BottomVal is never used.
         addAliasedAnnotation(MinLen.class, BOTTOMVAL);
 
-        // Only @Positive is aliased here (instead of the related lower bound checker annotations
-        // like @NonNegative, @IndexFor, etc.) because only @Positive provides useful
-        // information about @MinLen annotations. A similar annotation to @IntRangeFromPositive could
-        // be created for @NonNegative in the future.
+        // @Positive is aliased here because @Positive provides useful
+        // information about @MinLen annotations.
+        // @NonNegative and @GTENegativeOne are aliased similarly so
+        // that it's possible to overwrite a function annotated to return
+        // @NonNegative with, for instance, a function that returns an @IntVal(0).
         addAliasedAnnotation(
                 "org.checkerframework.checker.index.qual.Positive", createIntRangeFromPositive());
+        addAliasedAnnotation(
```

Comment:
You also need to add aliases for the aliases of `@NonNegative`.  For example, `@LengthOf`.